File: baselib-java/src/main/java/com/kuky/baselib_java/ResultCallBack.java
package com.kuky.baselib_java;

/**
 * @author Kuky
 */
public class ResultCallBack {

    public interface OnHttpRequestCallBack<T> {
        void onSuccess(T t);

        void onFail(String msg);
    }

    public interface onListCallBack<T> {
        void onSucceed(T t);

        void onEmpty();

        void onFailed();
    }

    public interface OnResultBack<T> {
        void onResult(T t);
    }
}

File: baselib/src/main/java/com/kuky/baselib/ResultCallBack.kt
package com.kuky.baselib

/**
 * @author Kuky
 */
class ResultCallBack {

    interface OnHttpResultBack<in T> {
        fun onSucceed(t: T)

        fun onFailed(msg: String)
    }

    interface OnListResultBack<in T> {
        fun onSucceed(t: T)

        fun onEmpty()

        fun onFailed()
    }

    interface OnResultBack<in T> {
        fun onResult(t: T)
    }
}